### PR TITLE
Feature/markets page

### PR DESCRIPTION
**Pull request by jmzwar**
Markets page + add a wallet tooltip to dashboard
Screen Shot 2023-05-12 at 7 01 35 pm
Screen Shot 2023-05-12 at 7 14 52 pm

### CODE REVIEW

File: v2/perps-v2/ui/src/components/Header.tsx [hunk 1]
```diff
@@ -1,29 +1,12 @@
...
 
 export const Header: FC = () => {
-  const { colorMode, toggleColorMode } = useColorMode();
-
-  useEffect(() => {
-    if (colorMode === 'light') {
-      toggleColorMode();
-    }
-  }, [colorMode, toggleColorMode]);
-
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/components/Header.tsx:26]: This is causing things to go grey

File: v2/perps-v2/ui/src/hooks/useMarkets.ts [hunk 2]
```diff
@@ -1,55 +1,147 @@
...
+            indexPrice = scale(wei(rawPriceInfo?.price), rawPriceInfo?.expo || 1);
+          }
+
+          const skewWithScale = wei(marketDetails?.marketSkew, 18, true).div(
+            wei(fundingParameters.skewScale, 18, true)
+          );
+
+          const markPrice = indexPrice.mul(wei(1).add(skewWithScale));
+
+          result.push({
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useMarkets.ts:121]: You're in a `.map` here, just return and you have your result array

File: v2/perps-v2/ui/src/hooks/useMarkets.ts [hunk 3]
```diff
@@ -1,55 +1,147 @@
...
-
-  return {
-    data,
-    loading,
-  };
+  const [state, setState] = useState<StateInterface>({ loading: true, data: null, error: null });
+  const client = useApolloClient();
+  const { upper, lower } = getDateRange(2, 3);
+
+  useEffect(() => {
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useMarkets.ts:45]: Miss react-query :(

File: v2/perps-v2/ui/src/hooks/useMarkets.ts [hunk 4]
```diff
@@ -1,55 +1,147 @@
...
+
+          return {
+            ...market,
+            percentageDifference,
+          };
+        });
+
+        const allMarketDetails = await perpsMarketDataContract.allMarketSummaries();
+
+        const fundingDetails = await Promise.all([
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useMarkets.ts:82]: It wouldn't be too hard to extract out some logic from usePosition and create a:
`const fetchMarketDetails = async (marketKeys: string[]) =>`

File: v2/perps-v2/ui/src/hooks/useMarkets.ts [hunk 5]
```diff
@@ -1,55 +1,147 @@
...
+          if (indexPrices && indexPrices[index]) {
+            const rawPriceInfo = indexPrices[index].getPriceUnchecked();
+            indexPrice = scale(wei(rawPriceInfo?.price), rawPriceInfo?.expo || 1);
+          }
+
+          const skewWithScale = wei(marketDetails?.marketSkew, 18, true).div(
+            wei(fundingParameters.skewScale, 18, true)
+          );
+
+          const markPrice = indexPrice.mul(wei(1).add(skewWithScale));
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useMarkets.ts:119]: Wanna extract out some of this logic? `calculateMarkPrice(price, skew, skewScale)`

**Review by bachstatter — COMMENTED** [review]
No markets showing up on preview url, is that expected?

File: v2/perps-v2/ui/src/hooks/useMarkets.ts (hunk 3 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useMarkets.ts:45]: hehe I don't

File: v2/perps-v2/ui/src/hooks/useMarkets.ts (hunk 4 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useMarkets.ts:82]: Yeah good point

File: v2/perps-v2/ui/src/hooks/useMarkets.ts (hunk 5 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useMarkets.ts:119]: We can

**Review by bachstatter — APPROVED** [review]
Approve if you wanna yolo

**Pull request merged**